src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -16,31 +23,110 @@ import { GAME_EVENTS } from '../game/GameController';
 // Temporary const declaration here to attach to the window AND use as internal property
 const aircraft = {};
 
-// TODO: this should be renamed to `AircraftCollection`
 /**
  * @class AircraftController
  */
 export default class AircraftController {
     /**
      * @constructor
+     * @for AircraftController
+     * @param aircraftTypeDefinitionList {array<object>}
+     * @param airlineController {AirlineController}
+     * @param navigationLibrary {NavigationLibrary}
      */
-    constructor() {
+    constructor(aircraftTypeDefinitionList, airlineController, navigationLibrary) {
+        if (!_isArray(aircraftTypeDefinitionList) || _isEmpty(aircraftTypeDefinitionList)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError('Invalid aircraftTypeDefinitionList passed to AircraftCollection. Expected and array but ' +
+                `received ${typeof aircraftTypeDefinitionList}`);
+        }
+
+        // TODO: this may need to use instanceof instead, but that may be overly defensive
+        if (!_isObject(airlineController) || !_isObject(navigationLibrary)) {
```

Comment:
this is me being lazy.  it is overly defensive in that this class is only instantiated once and in one spot and shouldn't _need_ to check that these two things are being passed correctly.  Using an instanceof check means a little more work for setting up tests because you need to use an actual instance instead of a stub or an empty object.

It does guarantee that a thing is what it's supposed to be, though.  But that's why I left the comment.  it should be good for now.